tests/typeahead-filter.js

```diff
@@ -29,18 +29,130 @@ describe('typeaheadFilter', function() {
 
   beforeEach(function() {
     this.$fixture.empty().append(
-      '<div class="filter js-typeahead-filter"  data-dataset="committees">' +
+      '<div class="js-typeahead-filter" data-dataset="committees">' +
         '<ul class="dropdown__selected"></ul>' +
-        '<input type="text" class="search-input--mini" name="commite_id">' +
+        '<input type="text" name="committee_id">' +
+        '<button type="button"></button>' +
       '</div>'
     );
 
-    this.typeaheadFilter = new TypeaheadFilter('.js-typeahead-filter', typeahead.datasets.committees);
+    this.typeaheadFilter = new TypeaheadFilter('.js-typeahead-filter', typeahead.datasets.committees, true);
   });
 
   it('should initialize', function() {
     var typeahead = this.typeaheadFilter.$body.find('.twitter-typeahead');
     expect(typeahead.length).to.equal(1);
   });
 
+  it('should set firstItem', function() {
+    this.typeaheadFilter.setFirstItem({}, {id: 'smith'});
+    expect(this.typeaheadFilter.firstItem.id).to.equal('smith');
+  });
+
+  it('should append checkbox and clear datum on typeahead:select', function() {
+    var appendCheckbox = sinon.spy(this.typeaheadFilter, 'appendCheckbox');
+    var datum = {
+      name: 'FAKE CANDIDATE',
+      id: '12345',
+      office: 'Senate'
+    };
+    this.typeaheadFilter.handleSelect({}, datum);
+
+    expect(appendCheckbox).to.have.been.calledWith({
+      name: 'committee_id',
+      label: 'FAKE CANDIDATE (12345)',
+      value: '12345',
+      id: 'committee_id-12345-checkbox'
+    });
+    expect(this.typeaheadFilter.datum).to.equal(null);
+
+    this.typeaheadFilter.appendCheckbox.restore();
+  });
+
+  it('should set this.datum on typeahead:autocomplte', function() {
+    this.typeaheadFilter.handleAutocomplete({}, {id: '12345'});
+    expect(this.typeaheadFilter.datum).to.deep.equal({id: '12345'});
+  });
+
+  it('should submit on enter', function() {
+    var handleSubmit = sinon.spy(this.typeaheadFilter, 'handleSubmit');
+    this.typeaheadFilter.handleKeypress({keyCode: 13});
+    expect(handleSubmit).to.have.been.calledWith({keyCode: 13});
+    this.typeaheadFilter.handleSubmit.restore();
+  });
```

Comment:
This is fine to write this way, but I prefer splitting the spy stuff into `beforeEach`/`afterEach`. I try to keep `it` blocks to only contain an action and the assertion. The rest is setup/cleanup.